### ISSUE TITLE
Pushing an invalid version name succeeds

### ISSUE THREAD

**Issue by avalanche123** [upbound/up#208]
### What happened?

Pushing a bad version silently fails - the package push works but it doesn't show up in the repository. Ideally, we'd fail fast.

<img width="1391" alt="image" src="https://user-images.githubusercontent.com/83289/180861819-036e4d63-5605-4058-b7e5-6d6276074b05.png">

Epic #

### How can we reproduce it?

Push any version that isn't semver compatible

### What environment did it happen in?

<img width="629" alt="image" src="https://user-images.githubusercontent.com/83289/180861995-2259a33d-1f20-4057-b009-9b9fb6fc5ba2.png">

**Comment by hasheddan** [comment]
> the package push works but it doesn't show up in the repository

@avalanche123 I would not expect the non-semver version to show up in the Marketplace, but are you saying it also does not show up in your Upbound console?

**Comment by avalanche123** [comment]
That's right, it is as if the push never happened, although I pushed multiple times

**Comment by hasheddan** [comment]
@avalanche123 ah right -- this is expected but I agree that it is quite opaque. The push actually does succeed, but we don't record it as a version that should be used given Crossplane's reliance on semantic versions for dependency resolution. If you did try to pull the image it would succeed though. I'll open an issue to track making this more obvious in the console and we can use this one to track potentially displaying a warning message about the behavior when performing the `push` 👍🏻
